rskj-core/src/test/java/co/rsk/peg/feeperkb/FeePerKbSupportImplTest.java

```diff
@@ -0,0 +1,191 @@
+        SignatureCache signatureCache = mock(SignatureCache.class);
+        Transaction tx = getTransactionFromAuthorizedCaller(signatureCache);
+
+        ABICallElection feePerKbElection = mock(ABICallElection.class);
+        when(feePerKbElection.vote(any(), any())).thenReturn(false);
+
+        AddressBasedAuthorizer authorizer = feePerKbConstants.getFeePerKbChangeAuthorizer();
+        when(storageProvider.getFeePerKbElection(authorizer)).thenReturn(feePerKbElection);
+
+        Coin feePerKbVote = Coin.valueOf(50_000L);
+
+        Integer actualResult = feePerKbSupport.voteFeePerKbChange(tx, feePerKbVote, signatureCache);
+
+        Integer expectedResult = FeePerKbResponseCode.UNSUCCESSFUL_VOTE.getCode();
+        assertEquals(expectedResult, actualResult);
+    }
+
+    @Test
+    void voteFeePerKbChange_successfulFeePerKbVote_shouldReturnSuccessfulFeeVotedResponseCode() {
+        SignatureCache signatureCache = mock(SignatureCache.class);
+        Transaction tx = getTransactionFromAuthorizedCaller(signatureCache);
+
+        ABICallElection feePerKbElection = mock(ABICallElection.class);
+        AddressBasedAuthorizer authorizer = feePerKbConstants.getFeePerKbChangeAuthorizer();
+        when(storageProvider.getFeePerKbElection(authorizer)).thenReturn(feePerKbElection);
+        when(feePerKbElection.vote(any(), any())).thenReturn(true);
+
+        Coin feePerKbVote = Coin.valueOf(50_000L);
+
+        Integer actualResult = feePerKbSupport.voteFeePerKbChange(tx, feePerKbVote, signatureCache);
+
+        Integer expectedResult = FeePerKbResponseCode.SUCCESSFUL_VOTE.getCode();
+        assertEquals(expectedResult, actualResult);
+    }
+
+    @Test
+    void voteFeePerKbChange_successfulFeePerKbChanged_shouldReturnSuccessfulFeeVotedResponseCode() {
+        final String SET_FEE_PER_KB_ABI_FUNCTION = "setFeePerKb";
+        final Coin feePerKbVote = Coin.valueOf(50_000L);
+
+        SignatureCache signatureCache = mock(SignatureCache.class);
+        Transaction tx = getTransactionFromAuthorizedCaller(signatureCache);
+
+        ABICallElection feePerKbElection = mock(ABICallElection.class);
+        when(feePerKbElection.vote(any(), any())).thenReturn(true);
+
+        AddressBasedAuthorizer authorizer = feePerKbConstants.getFeePerKbChangeAuthorizer();
+        when(storageProvider.getFeePerKbElection(authorizer)).thenReturn(feePerKbElection);
+
+        ABICallSpec feeVote = new ABICallSpec(SET_FEE_PER_KB_ABI_FUNCTION, new byte[][]{BridgeSerializationUtils.serializeCoin(feePerKbVote)});
+        when(feePerKbElection.getWinner()).thenReturn(Optional.of(feeVote));
+
+        Integer actualResult = feePerKbSupport.voteFeePerKbChange(tx, feePerKbVote, signatureCache);
+
+        Integer expectedResult = FeePerKbResponseCode.SUCCESSFUL_VOTE.getCode();
+        assertEquals(expectedResult, actualResult);
+    }
+
+    @Test
+    void voteFeePerKbChange_nullFeeThrows() {
```

Comment:
Is this the expected behaviour? Should we handle some way the case when the value is null?

Reply:
Yes, the feePerKb In-Parameter should be, not null.

Reply:
agree with marcos, we are not handling a null fee per kb. We should. But i think the expected behavior shouldn't be throwing a npe.